deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/IngestContentObjectsJobTest.java

```diff
@@ -0,0 +1,446 @@
+		when(work.getResource()).thenReturn(workResc);
+
+		String mainLoc = "pdf.pdf";
+		String mainMime = "application/pdf";
+		PID mainPid = addFileObject(workBag, mainLoc, mainMime);
+		String supLoc = "text.txt";
+		String supMime = "text/plain";
+		PID supPid = addFileObject(workBag, supLoc, supMime);
+
+		workBag.asResource().addProperty(Cdr.primaryObject,
+				model.getResource(mainPid.getRepositoryPath()));
+
+		job.closeModel();
+
+		when(repository.objectExists(eq(workPid))).thenReturn(true);
+		when(repository.objectExists(eq(mainPid))).thenReturn(true);
+
+		job.run();
+
+		// Check that the work object was retrieved rather than created
+		verify(repository).objectExists(eq(workPid));
+		verify(repository, never()).createWorkObject(any(PID.class));
+		verify(repository).getWorkObject(any(PID.class));
+
+		// Main file object should not be touched
+		verify(repository).objectExists(eq(mainPid));
+		verify(work, never()).addDataFile(eq(mainPid), any(InputStream.class),
+				anyString(), anyString(), anyString());
+		verify(repository, never()).getFileObject(eq(mainPid));
+
+		// Supplemental file should be created
+		verify(repository).objectExists(eq(supPid));
+		verify(repository, never()).getFileObject(eq(supPid));
+		verify(work).addDataFile(eq(supPid), any(InputStream.class), eq(supLoc),
+				eq(supMime), anyString());
+
+		// Ensure that the primary object still got set
+		verify(work).setPrimaryObject(mainPid);
+
+		verify(jobStatusFactory).setCompletion(eq(jobUUID), eq(2));
+		verify(jobStatusFactory).incrCompletion(eq(jobUUID), eq(1));
+	}
+
+	@Test
+	public void resumeIngestFileObjectAsWorkTest() {
+		// Mark the deposit as resumed
+		when(depositStatusFactory.isResumedDeposit(anyString())).thenReturn(true);
+
+		Model model = job.getWritableModel();
+		Bag depBag = model.createBag(depositPid.getRepositoryPath());
+
+		String fileLoc = "text.txt";
+		String fileMime = "text/plain";
+		PID filePid = addFileObject(depBag, fileLoc, fileMime);
+
+		when(repository.objectExists(eq(filePid))).thenReturn(true);
+
+		job.closeModel();
+
+		job.run();
+
+		// Check that the generated work was neither generated nor linked
+		verify(repository, never()).createWorkObject(any(PID.class), any(Model.class));
+		verify(destinationObj, never()).addMember(any(ContentObject.class));
+
+		// Only tick should be from preprocessing
+		verify(jobStatusFactory, never()).incrCompletion(eq(jobUUID), anyInt());
+		verify(jobStatusFactory).setCompletion(eq(jobUUID), eq(1));
+	}
+
+	/**
+	 * Check that no significant deposit behaviors change with a file object as
+	 * work deposit due to resume being turned on
+	 */
+	@Test
+	public void resumeIngestNoCompleteTest() {
+		// Mark the deposit as resumed
+		when(depositStatusFactory.isResumedDeposit(anyString())).thenReturn(true);
+
+		ingestFileObjectAsWorkTest();
+
+		// No preprocessing ticks
+		verify(jobStatusFactory).setCompletion(eq(jobUUID), eq(0));
+	}
+
+	@Test
+	public void permTest() {
```

Comment:
Whoops, didn't mean to leave that in, I was just testing to see how two different ways of checking for RDF.types performed to see if I could remove some code from the ingest job.  I'll take this out